lib/spack/spack/patch.py

```diff
@@ -119,7 +119,8 @@ def apply(self, stage):
             patch_stage.fetch()
             patch_stage.check()
             patch_stage.cache_local()
-            patch_stage.expand_archive()
+            if self.url.lower().endswith('.tar'):
+                patch_stage.expand_archive()
```

Comment:
@naromero77 

> A naive question. Does the current implementation handle other archives?

Yes - you can check `spack.util.compression` if you are interested in the details. In summary it currently support:
```python
# Supported archive extensions.
PRE_EXTS = ["tar"]
EXTS     = ["gz", "bz2", "xz", "Z", "zip", "tgz"]
```
> If expand_archive handles other archive types, e.g. *.zip it is simple to add.

I don't think it's a good idea to duplicate logic that prevents the expansion like what you are doing here. I still think it would be better if you can compress the patch and provide the compressed url somewhere (and that will work out of the box). 

Anyhow, if that's not possible, I don't like to let people down with deadlines so I baked #5197. Can you see if it works for your case? You should just add the keyword `expand=False` to the patch directive.